src/xdg/desktopfile.cpp

```diff
@@ -961,7 +962,7 @@ DesktopFile *DesktopFileCache::getDefaultApp(const QString &mimeType)
     for (const QString &mimeDir : const_cast<const QStringList &>(mimeDirsList)) {
         QString defaultsListPath = mimeDir + QStringLiteral("/mimeapps.list");
         if (QFileInfo::exists(defaultsListPath)) {
-            DesktopFile *defaultsFile = new DesktopFile();
+            auto defaultsFile = std::make_unique<DesktopFile>();
```

Comment:
Note that I've used make_unique that's available only since C++14, so either merge https://github.com/lirios/cmake-shared/pull/4 first or I can change it in C++11-compatible way.